registry/storage/driver/neofs/neofs.go

```diff
@@ -0,0 +1,887 @@
+	walletInfo := new(Wallet)
+
+	walletParams, ok := parameters[paramWallet].(map[interface{}]interface{})
+	if !ok {
+		return nil, fmt.Errorf("no wallet params provided")
+	}
+
+	walletInfo.Path, ok = walletParams[paramPath].(string)
+	if !ok {
+		return nil, fmt.Errorf("no path provided")
+	}
+
+	walletInfo.Password, ok = walletParams[paramPassword].(string)
+	if !ok {
+		return nil, fmt.Errorf("no password provided")
+	}
+
+	addressParam := walletParams[paramAddress]
+	if addressParam != nil {
+		if walletInfo.Address, ok = addressParam.(string); !ok {
+			return nil, fmt.Errorf("invalid address param")
+		}
+	}
+
+	return walletInfo, nil
+}
+
+func parsePeers(parameters map[string]interface{}) ([]*PeerInfo, error) {
+	poolParams, ok := parameters[paramPeers].(map[interface{}]interface{})
+	if !ok {
+		return nil, fmt.Errorf("no peers params provided")
+	}
+
+	var peers []*PeerInfo
+	for _, val := range poolParams {
+		peerInfo, ok := val.(map[interface{}]interface{})
+		if !ok {
+			return nil, fmt.Errorf("invalid peers params")
+		}
+
+		peer := new(PeerInfo)
+
+		peer.Address, ok = peerInfo[paramAddress].(string)
+		if !ok {
+			return nil, fmt.Errorf("invalid peer address")
+		}
+
+		weightParam := peerInfo[paramWeight]
+		if weightParam != nil {
+
+			switch weight := weightParam.(type) {
+			case int:
+				peer.Weight = float64(weight)
+			case float64:
+				peer.Weight = weight
+			default:
+				return nil, fmt.Errorf("invalid weight param")
+			}
+			if peer.Weight <= 0 {
+				peer.Weight = 1
+			}
+		}
+
+		priorityParam := peerInfo[paramPriority]
+		if priorityParam != nil {
+			if peer.Priority, ok = priorityParam.(int); !ok {
+				return nil, fmt.Errorf("invalid priority param")
+			} else if peer.Priority <= 0 {
+				peer.Priority = 1
+			}
+		}
+
+		peers = append(peers, peer)
+	}
+
+	return peers, nil
+}
+
+func parseTimeout(parameters map[string]interface{}, name string, defaultValue time.Duration) (time.Duration, error) {
+	timeoutValue := parameters[name]
+	if timeoutValue == nil {
+		return defaultValue, nil
+	}
+
+	switch val := timeoutValue.(type) {
+	case int:
+		return time.Duration(val), nil
+	case int64:
+		return time.Duration(val), nil
+	case string:
+		timeout, err := time.ParseDuration(val)
+		if err != nil {
+			return 0, fmt.Errorf("couldn't parse duration '%s': %w", val, err)
+		}
+		return timeout, nil
+	}
+
+	return 0, fmt.Errorf("invalid %s", name)
+}
+
+func parseUInt64(parameters map[string]interface{}, name string, defaultValue uint64) (uint64, error) {
+	expirationValue := parameters[name]
+	if expirationValue == nil {
+		return defaultValue, nil
+	}
+
+	switch val := expirationValue.(type) {
+	case int:
+		return uint64(val), nil
+	case int64:
+		return uint64(val), nil
+	}
+
+	return 0, fmt.Errorf("invalid %s", name)
+}
+
+// New constructs a new Driver with the given NeoFS params
+func New(params DriverParameters) (*Driver, error) {
```

Comment:
Do I understand correctly that the function cannot be named as `NewDriver` it must be `New`?

Reply:
No, it mustn't be. I just tried to follow to the format of other drivers (e.g. [s3](https://github.com/nspcc-dev/distribution/blob/main/registry/storage/driver/s3-aws/s3.go), [azure](https://github.com/nspcc-dev/distribution/blob/main/registry/storage/driver/azure/azure.go))